pkg/eventprocessor/eventfeed/impl/analytics.go

```diff
@@ -37,6 +39,19 @@ func (ef *EventFeed) fetchExtraBlockInfo(ctx context.Context) {
 						Msg("get extra block info")
 					return
 				}
+				newBlockMetric := telemetry.NewBlockMetric{
+					Version:            telemetry.NewBlockMetricV1,
+					ChainID:            int(ef.chainID),
+					BlockNumber:        blockNumber,
+					BlockTimestampUnix: block.Time,
+				}
+				if err := telemetry.Collect(ctx, newBlockMetric); err != nil {
+					ef.log.Error().
+						Err(err).
+						Int64("block_number", blockNumber).
+						Msg("capturing new block metric")
+					return
+				}
```

Comment:
We save first the new block information as a metric, and _then_ in the database. This is important since we do it in reverse and save the metric failed, we would miss sending the metric again (i.e: since the way we detect if a block information has been fetched, is by checking if it was inserted in `system_evm_blocks`).

In case that `InsertBlockExtraInfo` fails, this block will be retried and duplicate the metric persistence, but this is fine since in this case we'll dedupe at the analytics layer.